docs/csharp/whats-new/csharp-7-1.md

```diff
@@ -0,0 +1,176 @@
+'Properties'. Select the "Build" tab and click on the "Advanced" button. In the dropdown,
+select "C# latest minor version (latest)", or the specific version "C# 7.1"
+as shwon in the image below. The 'latest' value means you want to use the latest
+minor versio on the current machine. The 'C# 7.1' means that you want to use C# 7.1,
+even after newer minor versions are released.
+
+[Setting the language version](./csharp-7-1/media/advanced-build-settings.png)
+
+Alternatively, you can edit the "csproj" file and add or modify the
+following lines:
+
+```xml
+<PropertyGroup>
+  <LangVersion>latest</LangVersion>
+</PropertyGroup>
+```
+
+> [!NOTE]
+> If you use the Visual Studio IDE to update your csproj files, the IDE
+> creates separate nodes for each build configuration. You'll typically
+> set the value the same in all build configurations.
+
+Valid settings for "langVersion" element are:
+
+* ISO-1
+* ISO-2
+* 3
+* 4
+* 5
+* 6
+* 7
+* 7.1
+* default
+* latest
+
+The special strings "default" and "latest" resolve to the latest major
+and minor language versions installed on the build machine, respectively.
+
+This setting decouples installing new versions of the SDK and tools
+in your development environment from choosing to incorporate new language
+features in a project. You can install the latest SDK and tools on your
+build machine. Each project can be configured to use a specific version
+of the language for its build.
+
+## Async main
+
+An *async main* method enables you to use `await` in your `Main` method.
+Where previously you would need to write:
+
+```csharp
+static int Main()
+{
+    return DoAsyncWork().GetAwaiter().GetResult();
+}
+```
+
+You can know write:
+
+```csharp
+static async Task<int> Main()
+{
+    // This could also be replaced with the body
+    // DoAsyncWork, including its await expressions:
+    return await DoAsyncWork();
+}
+```
+
+If your program does not return an exit code, you can declare `Main` method
+that returns a `Task`:
+
+```csharp
+static async Task Main()
+{
+    await SomeAsyncMethod();
+}
+```
+
+You can read more about the details in the
+[async main](../programming-guid/main-and-command-args/index.md) topic
+in the programming guide.
+
+## Default value expressions
+
+When you initialize an object using a default value expression, you can
+omit the type from the default expression. Where you previously needed to
+type:
+
+```csharp
+Func<string, bool> whereClause = default(Func<string, bool>);
+```
+
+You can now write:
+
+```csharp
+Func<string, bool> whereClause = default;
+```
+
+There are many language constructs where default value expressions can make use
+of this new syntax. You can learn more in the topic on default value expressions
+in the topics on the [`default` keyword](../language-reference/keywords/default.md)
+and the topic on [default value expressions](../programming-guide/statements-expressions-operators/default-value-expressions.md).
```

Comment:
There's a lot of repetition in this paragraph, starting with an unspecified topic on default value expressions, with (plural) topics and a link to a single topic on the default keyword, and then the reference to default value expressions.